docs/user-guide/configuration.rst

```diff
@@ -12,6 +12,186 @@ Additionally, you can configure modules from the ``Artemis-modules-extra`` repos
 the configuration variables from https://github.com/CERT-Polska/Artemis-modules-extra/blob/main/extra_modules_config.py. The file to put them
 in (``.env``) and the syntax (``VARIABLE_NAME=VARIABLE_VALUE``) is the same as for the core Artemis configuration.
 
+Module Configuration
+-------------------
+
+The ``ModuleConfiguration`` class serves as the base for all module-specific configurations in Artemis. It provides a standardized way to handle module configurations with serialization, deserialization, and validation capabilities.
```

Comment:
I think we need to underscore, maybe in multiple places, that it's runtime configuration, for stuff you want to change during scanning (how aggressive the scanning is)